programs/bpf_loader/src/lib.rs

```diff
@@ -498,12 +494,10 @@ fn process_instruction_inner(
     let mut get_or_create_executor_time = Measure::start("get_or_create_executor_time");
     let executor = invoke_context
         .find_program_in_cache(program_account.get_key())
-        .ok_or(InstructionError::InvalidAccountData)?;
-
-    if executor.is_tombstone() {
-        return Err(Box::new(InstructionError::InvalidAccountData));
-    }
-
+        .ok_or_else(|| {
+            ic_logger_msg!(log_collector, "Program is not cached");
+            InstructionError::InvalidAccountData
```

Comment:
Why is the is_tombstone check not needed anymore here?

Reply:
It is redundant. See implementation of `is_tombstone()` [here](https://github.com/solana-labs/solana/blob/2f7a76760e365f76c0a16745ff87cdd77e706b6e/program-runtime/src/loaded_programs.rs#L330), which matches exactly the three cases below and also throws `InstructionError::InvalidAccountData`:
```
LoadedProgramType::FailedVerification(_)
| LoadedProgramType::Closed
| LoadedProgramType::DelayVisibility
```

That has nothing to do with the RBPF bump, so I can split it into a separate PR.

Reply:
I see, yes please split it out if you don't mind

Reply:
See #33145